test.py

```diff
@@ -28,6 +28,34 @@ def test_deparam(self):
             good = base + good
             self.assertEqual(url.parse(bad).deparam(['c']).utf8(), good)
 
+    def test_deparam_case_insensitivity(self):
+        '''Parameters to remove can be supplied in any case.'''
+        examples = [
+            ('?hELLo=2', ''),
+            ('?HELLo=2', '')
+        ]
+        base = 'http://testing.com/page'
+        for bad, good in examples:
+            bad = base + bad
+            good = base + good
+            self.assertEqual(url.parse(bad).deparam(['HeLlO']).utf8(), good)
+
+    def test_filter_params(self):
+        '''We can filter parameters.'''
+        def function(name, value):
```

Comment:
We don't have a use case for it now, and in the worst case users can hand over a functor that tracks state between invocations.

I'm content to let this grow organically, but it's not dissimilar from `filter` in `python` or `ruby`.